minedd/cli.py

```diff
@@ -7,16 +7,153 @@
+        '--llm',
+        type=str,
+        default='ollama/llama3.2:1b',
+        help='LLM model to use (default is llama3.2:1b)'
+    )
+
+    parser_embed.add_argument(
+        '--embedding_model',
+        type=str,
+        default='ollama/mxbai-embed-large:latest',
+        help='Embedding model to use (default is mxbai-embed-large:latest)'
+    )
+
+    parser_embed.add_argument(
+        "--output_dir",
+        type=str,
+        default="out",
+        help="Directory to save outputs (default: out)"
+    )
+
+    parser_embed.add_argument(
+        "--paper_directory",
+        type=str,
+        default="data/",
+        help="Directory containing paper files (default: data/)"
+    )
+
+    parser_embed.add_argument(
+        '--embeddings_filename',
+        type=str,
+        default='embeddings.pkl',
+        help='Filename for the embeddings pickle file (default: embeddings.pkl)'
+    )
+
+    parser_embed.add_argument(
+        '--augment_existing',
+        action='store_true',
+        help='Augment existing embeddings if they exist'
+    )
+
+    return parser_embed
+
+def add_query_args(parser_query):
+    parser_query.add_argument(
+        '--llm',
+        type=str,
+        default='ollama/llama3.2:1b',
+        help='LLM model to use (default is llama3.2:1b)'
+    )
+
+    parser_query.add_argument(
+        '--embedding_model',
+        type=str,
+        default='ollama/mxbai-embed-large:latest',
+        help='Embedding model to use (default is mxbai-embed-large:latest)'
+    )
+
+    parser_query.add_argument(
+        "--output_dir",
+        type=str,
+        default="out",
+        help="Directory to save outputs (default: out)"
+    )
+
+    parser_query.add_argument(
+        "--paper_directory",
+        type=str,
+        default="data/",
+        help="Directory containing paper files (default: data/)"
+    )
+
+    parser_query.add_argument(
+        '--embeddings',
+        type=str,
+        required=True,
+        help='Path to the embeddings pickle file'
+    )
+
+    parser_query.add_argument(
+        "--questions_file",
+        type=str,
+        help="Path to Excel file with questions"
+    )
+
+    parser_query.add_argument(
+        "--question",
+        type=str,
+        help="Single question to ask"
+    )
+
+    parser_query.add_argument(
+        "--max_retries",
+        type=int,
+        default=2,
+        help="Maximum number of retries for model loading (default: 2)"
+    )
+
+    return parser_query
+
+def embed_command(args):
+    """ Handle the embed command"""
+
+    emb_path = os.path.join(args.output_dir, args.embeddings_filename)
+
+    # Create the Embeddings object
+    engine = Embeddings(
+        model=args.llm,
+        embedding_model=args.embedding_model,
+        paper_directory=args.paper_directory,
+        output_embeddings_path=emb_path,
+        existing_docs=None
+    )
+
+    # Load existing embeddings if desired and provided
+    if args.augment_existing:
```

Comment:
Thanks for adding this functionality! augmenting embeddings might be really useful